<launch>

    <arg name="marker1Id"         default="582"/>
    <arg name="marker2Id"         default="26"/>
    <arg name="markerSize"        default="0.25"/>    <!-- in m -->
    <arg name="eye"               default="left"/>
    <arg name="dct_normalization" default="True" />
    <arg name="dct_filter_size"   default="2" />
    <arg name="marker1_frame"     default="marker_hand_frame" />
    <arg name="marker2_frame"     default="marker_object_frame" />
    <arg name="robot_name"        default="locobot" />
    
    <node pkg="aruco_ros" type="odometry" name="odometry" output="screen">    
        <remap from="/camera_info" to="/camera/color/camera_info" />
        <remap from="/image" to="/camera/color/image_raw" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size" value="$(arg markerSize)"/>
        <param name="marker_id1" value="$(arg marker1Id)"/>
        <param name="marker_id2" value="$(arg marker2Id)"/>
        <param name="normalizeImage" value="$(arg dct_normalization)"/>
        <param name="dct_components_to_remove" value="$(arg dct_filter_size)"/>
        <param name="parent_name" value="stereo_gazebo_$(arg eye)_camera_optical_frame"/>
        <param name="child_name1" value="$(arg marker1_frame)" />
        <param name="child_name2" value="$(arg marker2_frame)" />
        <param name="world_frame" value="$(arg robot_name)/odom" />
        <param name="child_frame" value="$(arg robot_name)/base_link" />
        <rosparam file="$(find aruco_ros)/cfg/marker_poses.yaml" command="load" />
    </node>

</launch>
